src/index.ts

```diff
@@ -3,7 +3,148 @@
 'use-strict';
 
 import {
+  ICell, ICodeCellViewModel, IMarkdownCellViewModel, IRawCellViewModel
+} from 'jupyter-js-cell';
+
+import {
+  IObservableList
+} from 'phosphor-observablelist';
+
+import {
   Widget
 } from 'phosphor-widget';
 
 import './index.css';
+
+
+/**
+ * The interactivity modes for a notebook.
+ */
+export
+enum NotebookMode {
+  /**
+   * Command mode is used for navigation and manipulation.
+   */
+  Command,
+
+  /**
+   * Edit mode is used for text and code editing.
+   */
+  Edit
+}
+
+
+
+/**
+ * An enum which describes which interface property has changed.
+ */
+export
+enum NotebookProperty {
+  DefaultMimetype,
+  DirtyIndicator,
+  Trusted,
+  Mode,
+  SelectedCell,
+}
+
+
+/**
+ * The arguments object emitted with the `stateChanged` signal.
+ */
+export
+interface INotebookChangedArgs<T> {
```

Comment:
@sccolbert is the generic typing needed here given that the signal declaration below passes `<any>`?